jedi/evaluate/context/function.py

```diff
@@ -63,11 +63,19 @@ def infer_function_execution(self, function_execution):
         """
         Created to be used by inheritance.
         """
-        yield_exprs = get_yield_exprs(self.evaluator, self.tree_node)
-        if yield_exprs:
-            return ContextSet(iterable.Generator(self.evaluator, function_execution))
+        is_coroutine = self.tree_node.parent.type == 'async_stmt'
+        is_generator = bool(get_yield_exprs(self.evaluator, self.tree_node))
+
+        if is_coroutine:
+            if is_generator:
+                return ContextSet(iterable.AsyncGenerator(self.evaluator, function_execution))
```

Comment:
I think here we have to check if it's Python 3.5. This basically means that you check for `if evaluator.python_version == (3, 5): return ContextSet()` (+ add a warning).

If we don't do that we end up with an AsyncGenerator in 3.5 that doesn't really work in that version.